server/server.go

```diff
@@ -0,0 +1,196 @@
+package server
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/gorilla/mux"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/Clever/cron-admin/db"
+	"github.com/robfig/cron"
+)
+
+var (
+	cronUpdateFields = []string{
+		"IsActive",
+		"Function",
+		"Workload",
+		"CronTime",
+		"TimeZone",
+		"Created",
+	}
+	errEmptyFunctionInput  = fmt.Errorf("Error: Must include non-empty function")
+	errMissingUpdateFields = fmt.Errorf("Error: Updates require you to include the following:  %s", strings.Join(cronUpdateFields, ", "))
+)
+
+func byteHandler(handler func(*http.Request) ([]byte, int, error)) func(http.ResponseWriter, *http.Request) {
+	return func(rw http.ResponseWriter, req *http.Request) {
+		data, statusCode, err := handler(req)
+		if err != nil {
+			fmt.Printf("Received handler err: %s\n", err)
+			rw.WriteHeader(statusCode)
+			rw.Write([]byte(err.Error()))
+		} else {
+			rw.WriteHeader(statusCode)
```

Comment:
looks like the call to `WriteHeader` is common to both blocks and can be placed before the `if`